Support for Chandrayaan2

oleg-alexandrov opened the pull request:
##  Support for Chandrayaan2TMC

This incorporates Adam's pull request https://github.com/DOI-USGS/ISIS3/pull/5767. It is rebased to latest, tests have been made to pass, and some adjustments were made to parsing some Chandrayaan2 xml files that lack some fields. 

This is not ready yet to be merged in. Support for Chandrayaan2 OHRC will be added on top.

## How Has This Been Validated?
<!--- All changes need to be validated to confirm that they produce -->
<!--- scientifically accurate results. -->
<!--- If your changes include any new algorithms or changes to existing algorithms, -->
<!--- please indicate any publications or references for them. -->
<!--- If you manually validated the changes please indicate -->
<!--- what data you used and how you checked the results. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- - [x] My code follows the code style of this project. -->
- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added my user impacting change to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document. <!--- NOTE: You can only have one changelog entry per PR, see https://github.com/DOI-USGS/ISIS3/blob/dev/CONTRIBUTING.md -->


## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

Commit:
Initial Chandrayaan2TmcCamera

Commit:
Chandrayaan2tmc updates and fixes

Commit:
Updated arrays to vectors to prevent over allocating stack for large images

acpaquette committed:
Added custom rotation for chandrayaan2

acpaquette committed:
Quoted elements in the chandrayaan2 isisimport template

acpaquette committed:
Added chandrayaan2 datadir translation

acpaquette committed:
Reverted chandrayaan1 M3 unittest

oleg-alexandrov committed:
Add Chandrayaan2OHRC.tpl template

oleg-alexandrov committed:
Add chandrayaan2 to rclone.conf

oleg-alexandrov committed:
Changes for Chandrayaan2OHRC

oleg-alexandrov committed:
More additions for chandra2

oleg-alexandrov committed:
Add Chandra2 OHR testcase

oleg-alexandrov committed:
Add Chandra OHRC test

oleg-alexandrov committed:
Remove print statement

oleg-alexandrov committed:
Add Chandrayaan2Ohrc

oleg-alexandrov committed:
Fix ck frame id for Chandrayaan2Ohrc

oleg-alexandrov committed:
Add tiny cub for testing handrayaan2Tmc

oleg-alexandrov committed:
Fix unit test

oleg-alexandrov committed:
Fix TestsIsisImportChandrayaan2 test

oleg-alexandrov committed:
Will write later a test for Chandrayaan2OhrcCamera

oleg-alexandrov committed:
Wipe for now the Ohrc camera

oleg-alexandrov committed:
Add Chandrayaan TMC support to CHANGELOG.md

oleg-alexandrov committed:
Undo some indentation issues, update the changelong

oleg-alexandrov committed:
Make the Chandra test faster

oleg-alexandrov committed:
Rm unused vars

oleg-alexandrov committed:
Add Chandrayaan2OhrcCamera

oleg-alexandrov committed:
Move data to separate file from code

oleg-alexandrov committed:
Add import test for Chandrayaan2 OHRC

oleg-alexandrov committed:
Linux Arm port (#5811)

* Linux Arm env

* Linux Arm note in env

* Fix compile issue

* Fix more compile errors

* Switch to nanoflann=1.4.2nanoflann=1.4.2

* Update the Linux Arm env with recently built packages

* nanoflann version adjustment

* Pin nanoflann version

oleg-alexandrov committed:
Make more fields in OHRC optional

Kelvinrr commented:
changelog conflict is preventing tests from running.

oleg-alexandrov commented:
I rebased this branch to latest and fixed the conflict.

oleg-alexandrov commented:
The build is failing with very strange errors, which like is due to something not being right in the cloud:

```
[Container] 2025/06/10 16:35:38.686147 Running command mkdir ${CODEBUILD_SRC_DIR}/build
[Container] 2025/06/10 16:35:38.727731 Running command cp ${CODEBUILD_SRC_DIR}/ISIS3/isis/IsisPreferences ${CODEBUILD_SRC_DIR}/ISIS3/isis/build/
cp: cannot create regular file '/codebuild/output/src2637739507/src/ISIS3/isis/build/': Not a directory

```

It looks like it failed to create the build directory. 

Another pull request unrelated to this is failing in the same place.

Kelvinrr commented:
@oleg-alexandrov might be from me messing with things, I can look at it.

Kelvinrr commented:
I patched it in the back, a new job should run in a bit.

Kelvinrr commented:
ignore these tests

```
	1308 - TempTestingFiles.UnitTestImagePolygonCross (Failed)
	1469 - DefaultCube.FunctionalTestCam2CamOffbody (Failed)
	2034 - Isisminer.IsisminerTestGisIntersect (Failed)
```

they are addressed in #5825 

But the new test seems to be failing.

oleg-alexandrov committed:
Fix Chandrayaan2Ohrc test

oleg-alexandrov commented:
Thanks. The new test was failing because is something I forgot to finish up. Addressed now. This branch is still work in progress, but I will keep an eye on how the tests are doing (I also rebased to latest).

oleg-alexandrov committed:
Fix some warnings

oleg-alexandrov commented:
Thanks. I haven't fully evaluated yet if the produced camera projects exactly where it should, as still having some issues on the CSM side, but I'll make another smaller pull request, if need be, for any extra work.
